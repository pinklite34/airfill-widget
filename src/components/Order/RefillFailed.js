import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'react-emotion';
import Button from '../UI/Button';

import OrderHeader from '../UI/OrderHeader';
import PaymentLayout from './PaymentLayout';
import Error from './error.svg';
import { orderProp, paymentStatusProp, fnProp } from '../../lib/prop-types';

const styles = {
  textContainer: css`
    display: block !important;
    line-height: 21px;
    margin-right: 48px;
  `,
  info: css`
    color: #777777;
    font-size: 14px;
  `,
  button: css`
    margin-top: 12px;
  `,
  link: css`
    text-decoration: none;
    color: #3e8fe4;
  `,
};

export default function RefillFailed(props) {
  const {
    order,
    paymentStatus: { failureData = {} },
    refundAddress,
    onReset,
  } = props;

  let mailto =
    'mailto:support@bitrefill.com?subject=' +
    encodeURIComponent(`Failed Order (ID ${order.id})`);

  if (!refundAddress) {
    mailto +=
      '&body=' +
      encodeURIComponent(`Hello,
My order (ID ${
        order.id
      }) failed to process. I'd like a refund to be sent to <Replace This With Your Refund Address>.

Thanks!`);
  }

  let text;
  let action;
  const needRefund =
    failureData && failureData.needRefund != null
      ? failureData.needRefund
      : order.needRefund;

  // Order is already refunded / not charged
  if (!refundAddress && (needRefund === false || order.refunded === true)) {
    text =
      'We have sent you an automatic refund. Please make sure your details are correct and try again!';
    action = (
      <Button
        onClick={onReset}
        text={{ id: 'button.sendrefill', children: 'Send another refill' }}
      />
    );
  } else if (refundAddress) {
    text =
      'We have sent you an automatic refund. You should receive it within a few minutes.';
    action = (
      <p>
        <a
          href={`https://live.blockcypher.com/btc/address/${refundAddress}/`}
          target="_blank"
          rel="noopener noreferrer">
          Click here to see how it&apos;s going
        </a>{' '}
        or <a href={mailto}>contact support@birefill.com</a>.
      </p>
    );
  } else {
    text =
      'Please use the button below to contact our support so that we can send you a refund.';
    action = (
      <Button
        href={mailto}
        text={{ id: 'button.contactsupport', children: 'Contact Support' }}
      />
    );
  }

  return (
    <div>
      <OrderHeader
        order={order}
        title="Delivery error"
        subtitle="The refill could not be delivered to the target account"
        icon={<Error />}
      />
      <PaymentLayout {...props}>
        <div>
          <div />
          <div className={styles.textContainer}>
            <span className={styles.info}>{text}</span>
            <br />
            <br />
            {action}
            <br />
            <br />
            <a
              className={`${styles.link} ${styles.help}`}
              href={`https://www.bitrefill.com/support/${order.orderId}/${
                order.payment.address
              }`}
              target="_blank"
              rel="noopener noreferrer">
              Click here for more information
            </a>
          </div>
        </div>
      </PaymentLayout>
    </div>
  );
}

RefillFailed.propTypes = {
  order: orderProp,
  paymentStatus: paymentStatusProp,
  refundAddress: PropTypes.string,
  onReset: fnProp,
};
