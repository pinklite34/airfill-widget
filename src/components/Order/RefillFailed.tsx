import * as React from 'react';

import { Order, PaymentStatus } from '../../lib/prop-types';
import Link from '../UI/Link';
import OrderHeader from '../UI/OrderHeader';
import OrderStatusButton from '../UI/OrderStatusButton';
import Text from '../UI/Text';
import Error from './error.svg';
import PaymentLayout from './PaymentLayout';

function getMailTo({ order, refundAddress }) {
  const subject = encodeURIComponent(`Failed Order (ID ${order.id})`);
  const body = encodeURIComponent(`Hello,

My order (ID ${
    order.id
  }) failed to process. I'd like a refund to be sent to ${refundAddress ||
    '<Replace This With Your Refund Address>'}.

Thanks!`);

  return `mailto:support@bitrefill.com?subject=${subject}&body=${body}`;
}

interface RefillFailedProps {
  order: Order;
  paymentStatus: PaymentStatus;
  refundAddress: string;
}

export default function RefillFailed(props: RefillFailedProps) {
  const {
    order,
    paymentStatus: { failureData = {} },
    refundAddress,
  } = props;

  const needRefund =
    failureData && failureData.needRefund != null
      ? failureData.needRefund
      : order.needRefund;

  const mailTo = getMailTo({ order, refundAddress });
  const isRefunded =
    !refundAddress && (needRefund === false || order.refunded === true);

  return (
    <React.Fragment>
      <OrderHeader
        order={order}
        title={{ id: 'order.failed.title', children: 'Delivery error' }}
        subtitle={{
          id: 'order.failed.subtitle',
          children: 'The refill could not be delivered to the target account',
        }}
        icon={<Error />}
      />

      <PaymentLayout {...props}>
        {order.errorMessage && (
          <Text error type="p" size="14px">
            {order.errorMessage}
          </Text>
        )}

        {isRefunded ? (
          <Text type="p" id="order.failed.info.check">
            We have sent you an automatic refund. Please make sure your details
            are correct and try again!
          </Text>
        ) : refundAddress ? (
          <Text type="p" id="order.failed.info.soon">
            We have sent you an automatic refund. You should receive it within a
            few minutes.
          </Text>
        ) : null}

        <OrderStatusButton
          order={order}
          text={
            !isRefunded && !refundAddress
              ? {
                  id: 'order.failed.refund',
                  children: 'Request a refund',
                }
              : null
          }
        />

        {refundAddress ? (
          <Text type="p" id="order.failed.transaction">
            <Link
              href={`https://live.blockcypher.com/btc/address/${refundAddress}/`}
            >
              Click here to see how it&apos;s going
            </Link>{' '}
            or <Link href={mailTo}>contact support@bitefill.com</Link>.
          </Text>
        ) : (
          <Link href={mailTo}>
            <Text type="link" id="order.failed.support">
              Click to email support
            </Text>
          </Link>
        )}
      </PaymentLayout>
    </React.Fragment>
  );
}

/* RefillFailed.propTypes = {
  order: orderProp,
  paymentStatus: paymentStatusProp,
  refundAddress: PropTypes.string,
};
 */
