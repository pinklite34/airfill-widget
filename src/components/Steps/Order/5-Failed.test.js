import React from 'react';
import RefillFailed from './5-Failed';
import renderer from 'react-test-renderer';

const props = {
  order: { id: '--ORDER_ID--' },
  onReset: jest.fn(),
  paymentStatus: {}
};

it('renders correctly', () => {
  const tree = renderer.create(<RefillFailed {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('supports auto-refunded orders', () => {
  const tree = renderer
    .create(<RefillFailed {...props} order={{ needRefund: false }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();

  const tree2 = renderer
    .create(<RefillFailed {...props} order={{ needRefund: true }} paymentStatus={{ failureData: {needRefund: false} }} />)
    .toJSON();
  expect(tree2).toMatchSnapshot();
});

it('supports refund address', () => {
  const tree = renderer
    .create(<RefillFailed {...props} refundAddress="--REFUND_ADDR--" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
