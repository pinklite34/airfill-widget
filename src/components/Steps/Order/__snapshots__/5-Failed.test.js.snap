// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Payment complete
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Refill sent
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="dIUkup"
    >
      X 
    </span>
    Delivery failed
  </h3>
  <p>
    For some reason we failed do deliver your refill. This can happen if you have typed the number incorrectly, if the number is not for a prepaid phone or if the operator you selected was not the right one.
  </p>
  <p>
    Please use the button below to contact our support so that we can send you a refund.
  </p>
  <a
    className="eFndht"
    href="mailto:support@bitrefill.com?subject=Failed%20Order%20(ID%20--ORDER_ID--)&body=Hello%2C%0AMy%20order%20(ID%20--ORDER_ID--)%20failed%20to%20process.%20I'd%20like%20a%20refund%20to%20be%20sent%20to%20%3CReplace%20This%20With%20Your%20Refund%20Address%3E.%0A%0AThanks!"
  >
    Contact Support
  </a>
</div>
`;

exports[`supports auto-refunded orders 1`] = `
<div>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Payment complete
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Refill sent
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="dIUkup"
    >
      X 
    </span>
    Delivery failed
  </h3>
  <p>
    For some reason we failed do deliver your refill. This can happen if you have typed the number incorrectly, if the number is not for a prepaid phone or if the operator you selected was not the right one.
  </p>
  <p>
    We have sent you an automatic refund. Please make sure your details are correct and try again!
  </p>
  <button
    className="eFndht"
    onClick={[Function]}
    type="button"
  >
    Send another refill
  </button>
</div>
`;

exports[`supports refund address 1`] = `
<div>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Payment complete
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="camDcq"
    >
      ✓ 
    </span>
    Refill sent
  </h3>
  <h3
    className="elengw"
  >
    <span
      className="dIUkup"
    >
      X 
    </span>
    Delivery failed
  </h3>
  <p>
    For some reason we failed do deliver your refill. This can happen if you have typed the number incorrectly, if the number is not for a prepaid phone or if the operator you selected was not the right one.
  </p>
  <p>
    We have sent you an automatic refund. You should receive it within a few minutes.
  </p>
  <p>
    <a
      href="https://live.blockcypher.com/btc/address/--REFUND_ADDR--/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Click here to see how it's going
    </a>
     or 
    <a
      href="mailto:support@bitrefill.com?subject=Failed%20Order%20(ID%20--ORDER_ID--)"
    >
      contact support@birefill.com
    </a>
    .
  </p>
</div>
`;
